apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V57_preadj_add_rbac_groups_and_roles.sql

```diff
@@ -0,0 +1,128 @@
+${logic.perms}      CREATE ROLE paca_migrator_role;
+${logic.perms}  END IF;
+${logic.perms}  GRANT ALL PRIVILEGES ON SCHEMA pre_adj TO paca_migrator_role;
+${logic.perms}  GRANT ALL PRIVILEGES ON ALL TABLES IN SCHEMA pre_adj TO paca_migrator_role;
+${logic.perms}  GRANT ALL PRIVILEGES ON ALL SEQUENCES IN SCHEMA pre_adj TO paca_migrator_role; -- curval, nextval, setval
+${logic.perms}  ALTER DEFAULT PRIVILEGES IN SCHEMA pre_adj GRANT ALL PRIVILEGES ON TABLES TO paca_migrator_role;
+${logic.perms}  ALTER DEFAULT PRIVILEGES IN SCHEMA pre_adj GRANT ALL PRIVILEGES ON SEQUENCES TO paca_migrator_role;
+${logic.perms}  
+${logic.perms}
+${logic.perms}  -- new paca groups: paca_analyst_group, paca_data_admins_group, paca_db_admins_group (read+write+alter)
+${logic.perms}  
+${logic.perms}  -- analyst group
+${logic.perms}  IF NOT EXISTS (SELECT FROM pg_catalog.pg_roles WHERE rolname = 'paca_analyst_group') THEN
+${logic.perms}      CREATE GROUP paca_analyst_group WITH
+${logic.perms}          NOSUPERUSER
+${logic.perms}          NOCREATEDB
+${logic.perms}          NOCREATEROLE
+${logic.perms}          NOINHERIT
+${logic.perms}          NOLOGIN
+${logic.perms}          NOREPLICATION
+${logic.perms}          NOBYPASSRLS;
+${logic.perms}  END IF;
+${logic.perms}  GRANT CONNECT ON DATABASE fhirdb TO paca_analyst_group;
+${logic.perms}  GRANT paca_reader_role TO paca_analyst_group;
+${logic.perms}  
+${logic.perms}  -- data admins group
+${logic.perms}  IF NOT EXISTS (SELECT FROM pg_catalog.pg_roles WHERE rolname = 'paca_data_admins_group') THEN
+${logic.perms}      CREATE GROUP paca_data_admins_group WITH
+${logic.perms}          NOSUPERUSER
+${logic.perms}          NOCREATEDB
+${logic.perms}          NOCREATEROLE
+${logic.perms}          NOINHERIT
+${logic.perms}          NOLOGIN
+${logic.perms}          NOREPLICATION
+${logic.perms}          NOBYPASSRLS;
+${logic.perms}  END IF;
+${logic.perms}  GRANT CONNECT ON DATABASE fhirdb TO paca_data_admins_group;
+${logic.perms}  GRANT paca_writer_role TO paca_data_admins_group;
+${logic.perms}  
+${logic.perms}  -- db admins group
+${logic.perms}  IF NOT EXISTS (SELECT FROM pg_catalog.pg_roles WHERE rolname = 'paca_db_admins_group') THEN
+${logic.perms}      CREATE GROUP paca_db_admins_group WITH
+${logic.perms}          NOSUPERUSER
+${logic.perms}          NOCREATEDB
+${logic.perms}          NOCREATEROLE
+${logic.perms}          NOINHERIT
+${logic.perms}          NOLOGIN
+${logic.perms}          NOREPLICATION
+${logic.perms}          NOBYPASSRLS;
+${logic.perms}  END IF;
+${logic.perms}  GRANT CONNECT ON DATABASE fhirdb TO paca_db_admins_group;
+${logic.perms}  GRANT paca_migrator_role TO paca_db_admins_group;
+${logic.perms}  
+${logic.perms}
+${logic.perms}  -- ensure current api/pipeline have the necessary permissions for zero downtime
+${logic.perms}
+${logic.perms}  -- grant reader to existing reader roles
```

Comment:
I think a more explicit comment here would be helpful that explains the fact that from here down we are just supporting on a temporary basis the current roles that were created out of band in the past (since these do not get created in any previous flyway scripts they could be confusing to folks wanting to know where these come from) and that a future migration will drop these roles and create a new set of permanent roles.

Reply:
Added.